shared/src/types/key/secp256k1.rs

```diff
@@ -316,7 +323,10 @@ impl<'de> Deserialize<'de> for Signature {
         struct ByteArrayVisitor;
 
         impl<'de> Visitor<'de> for ByteArrayVisitor {
-            type Value = [u8; libsecp256k1::util::SIGNATURE_SIZE];
+            // [`libsecp265k1::util::SIGNATURE_SIZE`] is for a traditional
+            // signature on this curve. For Ethereum, an extra byte is included
+            // that prevents malleability attacks.
+            type Value = [u8; libsecp256k1::util::SIGNATURE_SIZE + 1];
```

Comment:
The `libsecp256k1::util::SIGNATURE_SIZE + 1` and comment should be pulled out into a const and docstring e.g. `SIG_BYTES_LEN`, then whereever we are using `[u8; 65]`/`libsecp256k1::util::SIGNATURE_SIZE + 1` or similar we should use the const instead e.g. `[u8; SIG_BYTES_LEN]`.